File: problems/字符串/434.字符串中的单词数.cpp
#include <iostream>
#include <sstream>
#include <string>
using namespace std;

int countSegments(string s) {
    int count = 0;
    string word;
    istringstream iss(s);
    while (iss >> word) count++;
    return count;
}

int main() {
    // 测试用例
    cout << countSegments("Hello, my name is John") << endl;  // 预期: 5
    cout << countSegments("") << endl;                        // 预期: 0
    cout << countSegments("    ") << endl;                   // 预期: 0
    cout << countSegments("a") << endl;                      // 预期: 1
    return 0;
}

File: problems/字符串/434.字符串中的单词数.py
#!/usr/bin/env python3
"""
434. 字符串中的单词数
"""


def count_segments(s: str) -> int:
    return len([w for w in s.split() if w])


if __name__ == "__main__":
    # 测试用例
    print(count_segments("Hello, my name is John"))  # 预期: 5
    print(count_segments(""))                       # 预期: 0
    print(count_segments("    "))                  # 预期: 0
    print(count_segments("a"))                     # 预期: 1
